src/Orleans/Core/Exceptions.cs

```diff
@@ -170,6 +170,21 @@ protected GrainExtensionNotInstalledException(SerializationInfo info, StreamingC
     }
 
     /// <summary>
+    /// Signifies that an request was cancelled due to target silo death.
+    /// </summary>
+    [Serializable]
+    public class SilosDeathException : OrleansException
```

Comment:
I wander if this is a good name for this exception.
maybe something more general, like: `TemporalFailure` and the message string will be "SiloDeathException".
@sergeybykov , what do you think?